File: Java/Medium/WrongAnswer/delete-columns-to-make-sorted-ii_6.java
class Solution {
    public int minDeletionSize(String[] strs) {
        
        int n = strs.length;
        int m = strs[0].length();
        int[] dp = new int[m];
        int res = 0;
        for (int i = 0; i < m; i++) {
            dp[i] = 1;
            for (int j = 0; j < i; j++) {
                boolean isSorted = true;
                for (int k = 0; k < n; k++) {
                    if (strs[k].charAt(j) > strs[k].charAt(i)) {
                        isSorted = false;
                        break;
                    }
                }
                if (isSorted) {
                    dp[i] = Math.max(dp[i], dp[j] + 1);
                }
            }
            res = Math.max(res, dp[i]);
        }
        return m - res;
    }
}

File: Rust/Medium/Accepted/delete-columns-to-make-sorted-ii_6.rs
impl Solution {
    pub fn min_deletion_size(strs: Vec<String>) -> i32 {
        
        let mut sorted = vec![true; strs.len()];
        let mut res = 0;
        for i in 0..strs[0].len() {
            let mut is_sorted = true;
            for j in 1..strs.len() {
                if sorted[j] && strs[j].chars().nth(i).unwrap() < strs[j - 1].chars().nth(i).unwrap() {
                    is_sorted = false;
                    break;
                }
            }
            if !is_sorted {
                res += 1;
            } else {
                for j in 1..strs.len() {
                    if strs[j].chars().nth(i).unwrap() > strs[j - 1].chars().nth(i).unwrap() {
                        sorted[j] = false;
                    }
                }
            }
        }
        res
    }
}